pkg/grpc/utils.go

```diff
@@ -37,31 +37,44 @@ var log = ctrl.Log.WithName("util")
 
 // CreateGrpcConnection create a grpc connection with given port and address
 func CreateGrpcConnection(address string, port int, caCertPath string, certPath string, keyPath string) (*grpc.ClientConn, error) {
-	options := []grpc.DialOption{grpc.WithUnaryInterceptor(TimeoutClientInterceptor)}
-
+	var caCert, cert, key []byte
 	if caCertPath != "" && certPath != "" && keyPath != "" {
-		caCert, err := ioutil.ReadFile(caCertPath)
+		var err error
+		caCert, err = ioutil.ReadFile(caCertPath)
 		if err != nil {
 			return nil, err
 		}
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM(caCert)
-
-		clientCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		cert, err = ioutil.ReadFile(certPath)
 		if err != nil {
 			return nil, err
 		}
+		key, err = ioutil.ReadFile(keyPath)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I thought that would break some backward compatibility with secrue mode off. But I'm not sure.

Reply:
🙈 I forget the situation without secure mode. And the current codes still could not be compatible without secure mode, it will return an error when parsing cert-key pair with cert and key as `nil`.😥

I think the better way is determining using secure mode at first, if we do not need secure mode, we create a connection by another function without cert things. And if we need secure mode, then load the certs (from file or from secrets in kubernetes) as []byte, then create a connection by `CreateGrpcConnectionFromRaw`. 

And I think each switch or flag at least could be explicitly controlled by the user, for example, if we could not determine using secure mode or not, just add a command flag. After finished the basic functionality, then we could do something easy-to-use.

Reply:
Ah yes I forgot the part. About the not-secure mode, the original implementation (I mean before this pr) is to call `CreateGrpcConnection` with no cert files specified. The function `CreateGrpcConnection` is used somewhere else in chaosmesh too. I thought it would be better if we would keep this feature.

Reply:
You are right. I am trying to refactor that in this new feature PR, that's bad.